storage/2020-08-04/datalakestore/filesystems/properties_get.go

```diff
@@ -25,88 +23,46 @@ type GetPropertiesResponse struct {
+		OptionsObject: fileSystemOptions{},
+		Path:          fmt.Sprintf("/%s", fileSystemName),
 	}
 
-	resp, err := client.GetPropertiesSender(req)
+	req, err := c.Client.NewRequest(ctx, opts)
 	if err != nil {
-		result.Response = autorest.Response{Response: resp}
-		err = autorest.NewErrorWithError(err, "datalakestore.Client", "GetProperties", resp, "Failure sending request")
+		err = fmt.Errorf("building request: %+v", err)
 		return
 	}
 
-	result, err = client.GetPropertiesResponder(resp)
+	resp.HttpResponse, err = req.Execute(ctx)
 	if err != nil {
-		err = autorest.NewErrorWithError(err, "datalakestore.Client", "GetProperties", resp, "Failure responding to request")
-	}
-
-	return
-}
-
-// GetPropertiesPreparer prepares the GetProperties request.
-func (client Client) GetPropertiesPreparer(ctx context.Context, accountName string, fileSystemName string) (*http.Request, error) {
-	pathParameters := map[string]interface{}{
-		"fileSystemName": autorest.Encode("path", fileSystemName),
-	}
-
-	queryParameters := map[string]interface{}{
-		"resource": autorest.Encode("query", "filesystem"),
-	}
-
-	headers := map[string]interface{}{
-		"x-ms-version": APIVersion,
+		err = fmt.Errorf("executing request: %+v", err)
+		return
 	}
 
-	preparer := autorest.CreatePreparer(
-		autorest.AsHead(),
-		autorest.WithBaseURL(endpoints.GetDataLakeStoreEndpoint(client.BaseURI, accountName)),
-		autorest.WithPathParameters("/{fileSystemName}", pathParameters),
-		autorest.WithQueryParameters(queryParameters),
-		autorest.WithHeaders(headers))
+	if resp.HttpResponse != nil {
 
-	return preparer.Prepare((&http.Request{}).WithContext(ctx))
-}
-
-// GetPropertiesSender sends the GetProperties request. The method will close the
-// http.Response Body if it receives an error.
-func (client Client) GetPropertiesSender(req *http.Request) (*http.Response, error) {
-	return autorest.SendWithSender(client, req,
-		azure.DoRetryWithRegistration(client.Client))
-}
-
-// GetPropertiesResponder handles the response to the GetProperties request. The method always
-// closes the http.Response Body.
-func (client Client) GetPropertiesResponder(resp *http.Response) (result GetPropertiesResponse, err error) {
-	if resp != nil && resp.Header != nil {
-
-		propertiesRaw := resp.Header.Get("x-ms-properties")
+		propertiesRaw := resp.HttpResponse.Header.Get("x-ms-properties")
 		var properties *map[string]string
 		properties, err = parseProperties(propertiesRaw)
 		if err != nil {
 			return
 		}
 
-		result.Properties = *properties
-		result.NamespaceEnabled = strings.EqualFold(resp.Header.Get("x-ms-namespace-enabled"), "tru")
-	}
-
-	err = autorest.Respond(
-		resp,
-		client.ByInspecting(),
-		azure.WithErrorUnlessStatusCode(http.StatusOK),
-		autorest.ByClosing())
-	result.Response = autorest.Response{Response: resp}
+		resp.Properties = *properties
+		resp.NamespaceEnabled = strings.EqualFold(resp.HttpResponse.Header.Get("x-ms-namespace-enabled"), "tru")
```

Comment:
is this one a bug? (both before/after)

```suggestion
		resp.NamespaceEnabled = strings.EqualFold(resp.HttpResponse.Header.Get("x-ms-namespace-enabled"), "true")
```

Reply:
it was a bug - fixed now 👍